xdsl/dialects/arith.py

```diff
@@ -474,6 +474,95 @@ def from_mnemonic(operand1: Union[Operation, SSAValue],
         return Cmpi.get(operand1, operand2, arg)
 
 
+@irdl_op_definition
+class Cmpf(Operation):
+    """
+    The cmpf operation compares its two operands according to the float
+    comparison rules and the predicate specified by the respective attribute.
+    The predicate defines the type of comparison: (un)orderedness, (in)equality
+    and signed less/greater than (or equal to) as well as predicates that are
+    always true or false.  The operands must have the same type, and this type
+    must be a float type, or a vector or tensor thereof.  The result is an i1,
+    or a vector/tensor thereof having the same shape as the inputs. Unlike cmpi,
+    the operands are always treated as signed. The u prefix indicates
+    *unordered* comparison, not unsigned comparison, so "une" means unordered or
+    not equal. For the sake of readability by humans, custom assembly form for
+    the operation uses a string-typed attribute for the predicate.  The value of
+    this attribute corresponds to lower-cased name of the predicate constant,
+    e.g., "one" means "ordered not equal".  The string representation of the
+    attribute is merely a syntactic sugar and is converted to an integer
+    attribute by the parser.
+
+    Example:
+
+    %r1 = arith.cmpf oeq, %0, %1 : f32
+    %r2 = arith.cmpf ult, %0, %1 : tensor<42x42xf64>
+    %r3 = "arith.cmpf"(%0, %1) {predicate: 0} : (f8, f8) -> i1
+    """
+    name: str = "arith.cmpf"
+    predicate: OpAttr[AnyIntegerAttr]
+    lhs: Annotated[Operand, floatingPointLike]
+    rhs: Annotated[Operand, floatingPointLike]
+    result: Annotated[OpResult, IntegerType(1)]
+
+    @staticmethod
+    def _test_equality_of_operand_types(operand1: SSAValue,
+                                        operand2: SSAValue):
+        if operand1.typ == operand2.typ:
+          return True
+        return False
+
+    @staticmethod
+    def get(operand1: SSAValue | Operation,
```

Comment:
Can you maybe merge this one with the next one? Like `arg : int | str`.

Reply:
Good idea, I've done that